app/models/user.rb

```diff
@@ -4,13 +4,19 @@ class User < ApplicationRecord
   has_many :folders
   has_many :uploads, through: :folders
 
-  validates :first_name,        presence: true, uniqueness: true
-  validates :last_name,        presence: true, uniqueness: true
+  validates :first_name,      presence: true
+  validates :last_name,       presence: true
   validates :cellphone,       presence: true, uniqueness: true
   validates :email,           presence: true, uniqueness: true
   validates :password_digest, presence: true
 
+  after_create :create_root
+
   def root_folder
     folders.find_by(parent_id: nil)
   end
+
+  def create_root
```

Comment:
@AELSchauer I like the creation of the root folder being moved back into the model. I wasn't aware of the 'after_create' callback. It really cleans up the User controller.